config/config.go

```diff
@@ -58,6 +67,61 @@ type DataCache struct {
 	TTLSeconds int    `mapstructure:"ttl_seconds"`
 }
 
+// StoredRequests configures the backend used to store requests on the server.
+type StoredRequests struct {
+	// Files should be true if Stored Requests should be loaded from the filesystem.
+	Files bool `mapstructure:"filesystem"`
+	// Postgres should be non-nil if Stored Requests should be loaded from a Postgres database.
+	Postgres *PostgresConfig `mapstructure:"postgres"`
+}
+
```

Comment:
This seems odd that we have multiple options in the struct and not calling out the backend explicitly.

Reply:
did you have anything specific in mind?

I did start out with an explicit type, like this:

```
type StoredRequests struct {
  Type string `mapstructure:"type"

  Database string `mapstructure:"dbname"`
  Host     string `mapstructure:"host"`
  // ... other DB options here
}
```

but... that didn't actually help much, because `validate()` still has to make sure all the options exist if `type == 'postgres'`.

Is that what you meant, or did you have something else in mind?

Reply:
I was thinking more along the lines of

```
type StoredRequests struct {
  Type string `mapstructure:"type"
  Config string `mapstructure:"config"`
}
```

Where `Type` would be the backend and `Config` would contain the metadata associated with the type. Also there's probably a very small number of backends that folks would probably use (file, db, key-value-store, http?)....are we overbuilding by trying build support for all types of backends vs being more opinionated and saying (only use file or db backend types)?

Reply:
That explicit `Type` string sounds pretty sensible to me. I'll add that.

The `Config string` doesn't really, though. I looked through the DB drivers in Go, and even different DBs support different config options. A string type would:

- Require us to define and interpret a string format (semicolon-separated? Support for escaping?)
- Take even _more_ validation logic, since we'd have to unpack and inspect the string to make sure it has valid params.

Reply:
I was thinking conceptually something like:

```
type StoredRequests struct {
	Type   string   `json:"type"`
	StoredRequestsConfig Config `json:"config"`
}
```

We would define all the possible options in `StoredRequestsConfig` and allow them to be empty if not used for the `Type`. Considering we may not be supporting too many backends, this is fine.

```
type StoredRequestsConfig struct {
	Port int `json:"port,omitempty"`
	Username string `json:"username,omitempty"`
	Password string `json:"password,omitempty"`
	Foo string `json:"foo,omitempty"`
	Bar int `json:"bar,omitempty"`
}
```

A bit more dirty 😄 

```
type StoredRequests struct {
	Type   string   `json:"type"`
	Config map[string]interface{} `json:"config"`
}
```

Reply:
Validation logic would switch based on type to check for valid params.

Reply:
I just realized my suggestion is the same as what you had originally did!

Reply:
🤦‍♂️ ... you and me both. I saw your suggestion and it didn't even look familiar.

per the offline convo... I don't have any changes planned for this.

I can add a `Type` if you change your mind and think it's helpful... but it didn't sound like you were too worried about it anymore either.